dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/tcp/TCPUtils.java

```diff
@@ -0,0 +1,312 @@
+    public static BootstrapExtended createTCPBootstrapOutbound(NettyFrameworkImpl framework,
+            Map<String, Object> tcpOptions) throws NettyException {
+        BootstrapConfiguration config = new TCPConfigurationImpl(tcpOptions, false);
+        BootstrapExtended bs = new BootstrapExtended();
+        bs.group(framework.getChildGroup());
+        bs.channel(NioSocketChannel.class);
+        // apply the existing user config to the Netty TCP channel
+        bs.applyConfiguration(config);
+        ChannelInitializerWrapper tcpInitializer = new TCPChannelInitializerImpl(config, framework);
+        bs.setBaseInitializer(tcpInitializer);
+        return bs;
+    }
+
+    private static ChannelFuture open(NettyFrameworkImpl framework, AbstractBootstrap bootstrap,
+            final TCPConfigurationImpl config, String inetHost, int inetPort, ChannelFutureListener openListener,
+            final int retryCount) {
+
+        ChannelFuture oFuture = null;
+        if (inetHost.equals("*")) {
+            inetHost = NettyConstants.INADDR_ANY;
+        }
+        if (config.isInbound()) {
+            oFuture = ((ServerBootstrapExtended) bootstrap).bind(inetHost, inetPort);
+        } else {
+            oFuture = ((BootstrapExtended) bootstrap).connect(inetHost, inetPort);
+        }
+        final ChannelFuture openFuture = oFuture;
+        if (openListener != null) {
+            openFuture.addListener(openListener);
+        }
+        final String newHost = inetHost;
+
+        openFuture.addListener(future -> {
+            if (future.isSuccess()) {
+
+                // add new channel to set of active channels, and set a close future to
+                // remove it
+            	// Get parent and increment active connections
+                final Channel channel = openFuture.channel();
+
+                if(config.isInbound()) {
+                    if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                        Tr.debug(tc, "Adding new channel group for " + openFuture.channel());
+                    }
+                    // TODO: Should we use this or maybe the event loop of the channel?
+                    framework.getActiveChannelsMap().put(channel, new DefaultChannelGroup(GlobalEventExecutor.INSTANCE));
+                }else {
+                	framework.getOutboundConnections().add(channel);
+                }
+
+                // set common channel attrs
+                channel.attr(ConfigConstants.NAME_KEY).set(config.getExternalName());
+                channel.attr(ConfigConstants.HOST_KEY).set(newHost);
+                channel.attr(ConfigConstants.PORT_KEY).set(inetPort);
+                channel.attr(ConfigConstants.IS_INBOUND_KEY).set(config.isInbound());
+                
+                // Listener to stop channel on close
+                // This should just log that the channel stopped
+                channel.closeFuture().addListener(innerFuture -> logChannelStopped(channel));
```

Comment:
What if it is already closed before the method of the open event listener is run?
(opens and closes straight away)
If we are just logging the stopped event it does not matter much but we have
quiesce not completing defects in the old codebase due to missed decrements that are
probably due to code that is 99.9% watertight.